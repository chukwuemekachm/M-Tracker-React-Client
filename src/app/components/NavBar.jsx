import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export class NavBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      size: '',
    };
    this.toggleMenu = this.toggleMenu.bind(this);
  }

  toggleMenu() {
    let { size } = this.state;
    size = size === 'none' ? 'block' : 'none';
    this.setState({ size });
  }

  render() {
    const { size } = this.state;
    const { authenticated } = this.props;
    return (
      <nav>
        <ul className="ch-nav ch-card">
          <li>
            <h2>
              <Link to="/">
                <i className="icon ion-md-construct" />
                {' '}
                Maintenance Tracker
              </Link>
              <i className="icon ion-md-menu" id="menu" onClick={this.toggleMenu} />
            </h2>
          </li>
          <div className="ch-nav-right" id="nav" style={{ display: size }}>
            <li>
              {
                authenticated
                  ? (
                    <Link to="/">
                      Logout
                    </Link>
                  )
                  : (
                    <Link to="/login">
                      Login
                    </Link>
                  )
              }
            </li>
          </div>
        </ul>
      </nav>
    );
  }
}

NavBar.propTypes = {
  authenticated: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  authenticated: state.auth.authenticated,
});

export default connect(mapStateToProps, null)(NavBar);
