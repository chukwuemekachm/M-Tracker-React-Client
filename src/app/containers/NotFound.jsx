import React from 'react';
import DefaultNavBar from '../components/NavBar';

/**
 * @description The Not found component
 *
 * @returns {object}
 */
const NotFound = () => (
  <div className="ch-aside">
    <DefaultNavBar />
    <div
      className="ch-row"
      style={{
        backgroundColor: 'rgba(0, 0, 0, 0.7)', color: 'white', width: '100%', minHeight: '100%', position: 'fixed',
      }}
    >
      <div className="ch-col-6">
        <h1 style={{ fontSize: '8em' }}>
          404
        </h1>
        <p>
          Oops, the page you are looking for was not found.
        </p>
      </div>
    </div>
  </div>
);

export default NotFound;
