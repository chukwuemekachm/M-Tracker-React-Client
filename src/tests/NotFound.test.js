import React from 'react';
import { shallow } from 'enzyme';
import NotFound from '../app/containers/NotFound';
import NavBar from '../app/components/NavBar';

describe('Tests 404 NotFound component', () => {
  it('should render without errors', () => {
    const wrapper = shallow(<NotFound />);
    expect(wrapper.find(NavBar)).toHaveLength(1);
    expect(wrapper.find('div')).toHaveLength(3);
    expect(wrapper.find('h1')).toHaveLength(1);
    expect(wrapper.find('p')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
