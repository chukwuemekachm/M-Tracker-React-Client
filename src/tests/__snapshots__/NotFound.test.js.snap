// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests 404 NotFound component should render without errors 1`] = `
<div
  className="ch-aside"
>
  <Connect(NavBar) />
  <div
    className="ch-row"
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.7)",
        "color": "white",
        "minHeight": "100%",
        "position": "fixed",
        "width": "100%",
      }
    }
  >
    <div
      className="ch-col-6"
    >
      <h1
        style={
          Object {
            "fontSize": "8em",
          }
        }
      >
        404
      </h1>
      <p>
        Oops, the page you are looking for was not found.
      </p>
    </div>
  </div>
</div>
`;
